crates/matrix-sdk-base/src/client.rs

```diff
@@ -1173,6 +1180,15 @@ impl BaseClient {
             }
         }
 
+        for (room_id, pinned_event_ids) in changes.pinned_events.clone() {
+            if let Some(room) = self.store.room(&room_id) {
+                room.pinned_event_ids.update(|ids| {
+                    ids.clear();
+                    ids.append(&mut pinned_event_ids.clone());
```

Comment:
:crab: :teacher: 

- [Vec::extend](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.extend) can take an iterator of the element type (that's `ids`)
- and a `Vec` also has an `IntoIter` implementation, so it can be used as the parameter for `extend` (here I'm talking about `pinned_event_ids`).

Combining the two, that gives you a way to avoid the cloning:

```suggestion
                    ids.extend(pinned_event_ids);
```

Reply:
I had to use `Vector::extend_from_slice` in some places instead, but it's done in https://github.com/matrix-org/matrix-rust-sdk/pull/3747/commits/8c43c7523b37a595cc9b6cddcf22da78d560c46b